src/components/Trade/TradeInput.tsx

```diff
@@ -297,35 +299,54 @@ export const TradeInput = () => {
     const minLimit = `${bnOrZero(quote?.minimum).decimalPlaces(6)} ${quote?.sellAsset.symbol}`
 
     if (!wallet) return 'common.connectWallet'
+    if (
+      sellTradeAsset?.asset?.chainId === undefined ||
```

Comment:
we don't need to check for `.chainId` on the `asset` here - the `asset` is an object which is truthy.

you can just do

`if (!sellTradeAsset?.asset || !buyTradeAsset?.asset) ...`

Reply:
@0xdef1cafe Regarding this check, it seems that it introduced a small regression, for an asset that is not supported by the wallet the button keeps displaying "Loading..." as `sellTradeAsset.asset`/`buyTradeAsset.asset` never get set in this case apparently. For example you can try selecting Bitcoin as an asset using MetaMask. 

Would it be possible to rather check if the attempt to load the asset is finished, display "Loading..." until it's done, and if it's done start showing the potential error messages?